build_modules/lib/src/module_library.dart

```diff
@@ -70,16 +96,27 @@ class ModuleLibrary {
           directive.configurations.isNotEmpty) {
         conditionalDirectiveConfigurations = directive.configurations;
       }
-      deps.add(linkedId);
       if (conditionalDirectiveConfigurations != null) {
-        deps.addAll(conditionalDirectiveConfigurations
-            .map((c) => Uri.parse(c.uri.stringValue))
-            .where((u) => u.scheme != 'dart')
-            .map((u) => AssetId.resolve(u.toString(), from: id)));
+        var conditions = <String, AssetId>{r'$default': linkedId};
+        for (var condition in conditionalDirectiveConfigurations) {
+          if (Uri.parse(condition.uri.stringValue).scheme == 'dart') {
+            throw 'Unsupported conditional import of '
```

Comment:
Why is this unsupported?

Reply:
I don't think it has any value (you can't use things from them anyways) - tracking it would have meant tracking sdk deps per platform which complicates things unnecessarily.

Reply:
> you can't use things from them anyways

I don't understand, this seems valid:

```dart
import 'fake_io.dart'
  if (dart.library.io) 'dart:io';
```

Reply:
Hmm I guess in that sense you could use it, I hadn't considered the fake io case but I don't think that is a common use case anyways?

My thoughts regarding this were that the different libraries don't have any overlapping types, so directly doing this would not have any value:

```
import 'default.dart'
  if (dart.library.io) 'dart:io'
  if (dart.library.html) 'dart:html';
```

Reply:
yeah I think the fake_io, fake_html thing is going to be a real use case...

I agree that there isn't much value in tracking it, and we can probably assume that they are "correct" in that you won't have something like `if (dart.library.io) 'dart:html'` - and let the failure be late if you do.

Is it OK to ignore them instead of throw here?

Reply:
I don't actually think the fake_io case would be common though - in what scenario would that actually be a good pattern?

Primarily this feature is meant to be used by libraries I think, where they are going to want some real implementation and not just fall back on a fake. I don't think it is particularly cumbersome to simply force people to come up with a more generic api and have their actual platform specific imports hidden behind a conditional import.

In theory you could use a fake within a test but I don't understand the value that the feature provides there since you are typically only running on one platform or always running with a fake/mock.